src/problem/constr/constr_BOUND.c

```diff
@@ -402,6 +405,230 @@ void CONSTR_BOUND_analyze_step(Constr* c, Bus* bus, BusDC* busdc, int t) {
 				   t);
     }
   }
+  
+  // VSC Converters
+  for (conv = BUS_get_vsc_conv(bus); conv != NULL; conv = CONVVSC_get_next_ac(conv)) {
+    
+    // Active Power (P)
+    if (CONVVSC_has_flags(conv,FLAG_VARS,CONVVSC_VAR_P)) {
+      index = CONVVSC_get_index_Q(conv,t);
+      MAT_set_i(G,index,index);
+      MAT_set_j(G,index,index);
+      MAT_set_d(G,index,1.);
+      if (CONVVSC_has_flags(conv,FLAG_BOUNDED,CONVVSC_VAR_P)) {
+      	VEC_set(u,index,CONVVSC_get_P_max(conv));
+      	VEC_set(l,index,CONVVSC_get_P_min(conv));
+      }
+      else {
+      	VEC_set(u,index,CONVVSC_INF_P);
+      	VEC_set(l,index,-CONVVSC_INF_P);
+      }
+      
+      // Row info
+      CONSTR_set_G_row_info_string(c,
+  				   index,
+  				   "vsc converter",
+  				   CONVVSC_get_index(conv),
+  				   "active power",
+  				   t);
+    }
+    
+    // Reactive Power (Q)
+    if (CONVVSC_has_flags(conv,FLAG_VARS,CONVVSC_VAR_Q)) {
+      index = CONVVSC_get_index_Q(conv,t);
+      MAT_set_i(G,index,index);
+      MAT_set_j(G,index,index);
+      MAT_set_d(G,index,1.);
+      if (CONVVSC_has_flags(conv,FLAG_BOUNDED,CONVVSC_VAR_Q)) {
+      	VEC_set(u,index,CONVVSC_get_Q_max(conv));
+      	VEC_set(l,index,CONVVSC_get_Q_min(conv));
+      }
+      else {
+      	VEC_set(u,index,CONVVSC_INF_Q);
+      	VEC_set(l,index,-CONVVSC_INF_Q);
+      }
+      
+      // Row info
+      CONSTR_set_G_row_info_string(c,
+  				   index,
+  				   "vsc converter",
+  				   CONVVSC_get_index(conv),
+  				   "reactive power",
+  				   t);
+    }
+    
+    // DC Power (P_dc)
+    if (CONVVSC_has_flags(conv,FLAG_VARS,CONVVSC_VAR_PDC)) {
+      index1 = CONVVSC_get_index_P_dc(conv,t);
+      index2 = CONVVSC_get_index_i_dc(conv,t);
+      
+      MAT_set_i(G,index1,index1);
+      MAT_set_j(G,index1,index1);
+      MAT_set_d(G,index1,1.);
+      MAT_set_i(G,index2,index2);
+      MAT_set_j(G,index2,index2);
+      MAT_set_d(G,index2,1.);
+      
+      VEC_set(u,index1,CONVVSC_INF_PDC);
+      VEC_set(l,index1,-CONVVSC_INF_PDC);
+      VEC_set(u,index2,CONVVSC_INF_PDC);
+      VEC_set(l,index2,-CONVVSC_INF_PDC);
+      
+      // Row info
+      CONSTR_set_G_row_info_string(c,
+  				   index1,
+  				   "vsc converter",
+  				   CONVVSC_get_index(conv),
+  				   "dc power",
+  				   t);
+      CONSTR_set_G_row_info_string(c,
+  				   index2,
+  				   "vsc converter",
+  				   CONVVSC_get_index(conv),
+  				   "dc current",
+  				   t);
+    }
+  }
+  
+  // FACTS
+  for (facts = BUS_get_facts_k(bus); facts != NULL; facts = FACTS_get_next_k(facts)) {
+    
+    // Series Voltage Magnitude (v_mag_s)
+    if (FACTS_has_flags(facts,FLAG_VARS,FACTS_VAR_VMAG_S)) {
+      index = FACTS_get_index_v_mag_s(facts,t);
+      MAT_set_i(G,index,index);
+      MAT_set_j(G,index,index);
+      MAT_set_d(G,index,1.);
+      if (FACTS_has_flags(conv,FLAG_BOUNDED,FACTS_VAR_VMAG_S)) {
```

Comment:
This should not compile, or if FACTS_has_flags uses void* then it would cause a segfault at runtime. My guess is that it is not covered by any of the unittests. No worries, I am making updates to the PR branches of all involved packages. I will make a summary of all my changes at the end

Reply:
@ttinoco Thanks. Let me know if you need any input from me.  I'll take a look at your changes once complete.  Sorry that I didn't make sure to test it on a case with FACTS.